File: Observer Pattern/CSharp/WeatherData/CurrentConditionsDisplay.cs
using System;

namespace WeatherData 
{
    public class CurrentConditionsDisplay : Observer.IObserver, Observer.IDisplayElement 
    {
        private float temperature;
        private float humidity;
        private WeatherData weatherData;

        public CurrentConditionsDisplay(WeatherData weatherData)
        {
            this.weatherData = weatherData;
            this.weatherData.OnRegister(this);
        }

        public void OnUpdate(float temperature, float humidity, float pressure) 
        {
            this.temperature = temperature;
            this.humidity = humidity;
            OnDisplay();
        }

        public void OnDisplay()
        {
            Console.WriteLine("현재 상태: 온도 " + temperature + "F, 습도 " + humidity + "%");
        }
    }
}

File: Patterns/Observer Pattern/Cpp/WeatherData/CurrentConditionsDisplay.cpp
#include <iostream>
#include "Observer/IObserver.cpp"
#include "Observer/IDisplayElement.cpp"
#include "WeatherData.cpp"

using namespace std;

class CurrentConditionsDisplay : public IObserver, IDisplayElement {
	float temperature;
	float humidity;
	WeatherData* weatherData;

public:
	CurrentConditionsDisplay(WeatherData* weatherData) {
		this->weatherData = weatherData;
		this->weatherData->registerObserver(this);
	}

	void update(float temperature, float humidity, float pressure) {
		this->temperature = temperature;
		this->humidity = humidity;
		display();
	}

	void display() {
		cout << "현재 상태: 온도 " << this->temperature << "F, 습도 " << this->humidity << "%\n";
	}
};
